sleap_nn/data/providers.py

```diff
@@ -0,0 +1,29 @@
+"""Handle importing of sleap data."""
+import torchdata.datapipes.iter as dp
+import lightning.pytorch as pl
+
+import sleap_io as sio
+
+
+class LabelsReader(dp.IterDataPipe):
+    """Datapipe for reading frames from Labels object."""
+
+    def __init__(self, labels: sio.Labels):
+        """Initialize LabelsReader.
+
+        Args:
+            labels: sleap_io.Labels object that contains LabeledFrames that will be
+                accessed through a torchdata DataPipe
+        """
+        self.labels = labels
+
+    @classmethod
+    def from_filename(cls, filename: str):
+        """Create LabelsReader from a .slp filename."""
+        labels = sio.load_slp(filename)
+        return cls(labels)
+
+    def __iter__(self):
+        """Return labeledframe sample."""
```

Comment:
Consider documenting the returned data of each element (see SLEAP's [`LabelsReader.make_dataset()` docstring](https://github.com/talmolab/sleap/blob/1c2be11400481ad2d4f3805010f10e65ff2eb0d3/sleap/nn/data/providers.py#L152-L185) for reference).

Since this is a data pipeline, we might want to convert this to an "example" dictionary with pure primitives (tensors/arrays, primitive scalars, etc) with the relevant metadata pulled out of the `sleap_io.LabeledFrame` object so that we don't have to pass that instance around.